### PR TITLE
Refactor vmware_preprovision method for VM Provisioning.

### PR DESCRIPTION
**Pull request by pkomanek**
Method refactoring and test creation.

This PR is based on the issue below.
ManageIQ/manageiq#12038

@miq-bot add_label refactoring
@miq-bot assign @tinaafitz

### CODE REVIEW

**Comment by lfu** [comment]
Please address rubocop comment.

File: spec/content/automate/ManageIQ/Infrastructure/VM/Provisioning/StateMachines/Methods.class/__methods__/vmware_preprovision_spec.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      current_object = Spec::Support::MiqAeMockObject.new
+      current_object.parent = root_object
+      service.object = current_object
+    end
+  end
+
+  def build_notes
+    res = "Owner: #{prov_opt[:owner_first_name]} #{prov_opt[:owner_last_name]}\n"\
+          "Email: #{prov_opt[:owner_email]}\n"\
+          "Source: #{template.name}"
```
Comment (lfu) [inline spec/content/automate/ManageIQ/Infrastructure/VM/Provisioning/StateMachines/Methods.class/__methods__/vmware_preprovision_spec.rb]: Could use ```%()``` here.

File: spec/content/automate/ManageIQ/Infrastructure/VM/Provisioning/StateMachines/Methods.class/__methods__/vmware_preprovision_spec.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      @set_vlan = true
+      main
+    end
+
+    expect(ae_service.root["miq_provision"].get_option(:vlan)).to eq("VM Network")
+  end
+
+  it 'sets the folder' do
+    v_owning_datacenter = "v_owning_datacenter"
+    allow_any_instance_of(MiqAeMethodService::MiqAeServiceMiqTemplate).to receive(:v_owning_datacenter).and_return(v_owning_datacenter)
```
Comment (lfu) [inline spec/content/automate/ManageIQ/Infrastructure/VM/Provisioning/StateMachines/Methods.class/__methods__/vmware_preprovision_spec.rb]: Could you try this:
```
svc_template = MiqAeMethodService::MiqAeServiceMiqTemplate.find(template.id)
allow(ae_service.root["miq_provision"]).to receive(:vm_template).and_return(svc_template)
allow(svc_template).receive(:v_owning_datacenter).and_return(v_owning_datacenter)
```

**Comment by djberg96** [comment]
@tinaafitz look ok?

**Pull request merged**